e2e-tests/cypress/tests/integration/channels/enterprise/metrics/license_spec.js

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Stage: @prod
+// Group: @channels @enterprise @metrics @not_cloud
+
+describe('Metrics > License', () => {
+    before(() => {
+        cy.shouldNotRunOnCloudEdition();
+        cy.apiUpdateConfig({
+            MetricsSettings: {
+                Enable: true,
+            },
+        });
+    });
+
+    const checkMetrics = (expectedStatusCode) => {
+        cy.apiGetConfig().then(({config}) => {
+            const baseURL = new URL(Cypress.config('baseUrl'));
+            baseURL.port = config.MetricsSettings.ListenAddress.replace(/^.+:/, '');
+            baseURL.pathname = '/metrics';
+
+            Cypress.log({name: 'Metrics License', message: `Checking metrics at ${baseURL.toString()}`});
+            cy.request({
+                headers: {'X-Requested-With': 'XMLHttpRequest'},
+                url: baseURL.toString(),
+                method: 'GET',
+                failOnStatusCode: false,
+            }).then((response) => {
+                expect(response.status).to.equal(expectedStatusCode);
+            });
+        });
+    };
+
+    it('should enable metrics in BUILD_NUMBER == dev environments regardless of having a license', () => {
+        cy.apiGetConfig(true).then(({config}) => {
+            if (config.BuildNumber !== 'dev') {
+                Cypress.log({name: 'Metrics License', message: `Skipping test since BUILD_NUMBER = ${config.BuildNumber}`});
+                return;
+            }
+
+            cy.apiDeleteLicense();
+            checkMetrics(200);
+
+            cy.apiRequireLicense();
+            checkMetrics(200);
```

Comment:
Understood, thank you for clarifying the nuances! I'll rework this as per the current limitations. I wonder if we could leverage the service environment to "safely" share a developer license that we don't care about risking exposure? That would sidestep the issues above with the slight caveat that we'd be testing "non-production" in a very small number of situations (all of which actually apply perfectly to a testing environment, e.g. not hitting Rudder). A smoketest that used a production license could validate the rest.

cc @marianunez for thoughts?

Reply:
I can confirm from an `/e2e-test` run that we're already using the `test` service environment:
![CleanShot 2023-10-19 at 14 46 52@2x](https://github.com/mattermost/mattermost/assets/1023171/3180fdd6-eaee-4ba8-b47e-71a3576b54d8)

@saturninoabril, does this mean we can dispense with concerns around leaking license and even go so far as to check in a set of licenses for various situations that we can upload and delete at will?

Reply:
If all are good with test licenses potentially leaking and no issue with it, then it's great as it will unblock test like this and others as well. I'll go ahead and enable in the CI.
Thanks @lieut-data and @marianunez for confirming!

Reply:
cc @iyampaul & @esarafianou, as per the above, our plan is to check in some test licenses that unlock a more holistic testing strategy, knowing full well that these may "leak". The licenses are only useful if `ServiceEnvironment == test`, but I admit there are few "consequences" for forcing such a service environment. That being said, I note this is "the GitLab way". Any thoughts or concerns?

Reply:
I'm not overly worried about this; the people using a leaked token for prod will be the same ones who will try to find other means of circumventing our licensing/auth method anyways. Regardless of them being able to unlock EE features they still won't get support from us if they need it, and will need to re-apply their "crack" each update.